GitUI/BranchTreePanel/BaseRefTree.cs

```diff
@@ -80,5 +84,56 @@ internal override void UpdateVisibility()
 
             base.UpdateVisibility();
         }
+
+        protected IEnumerable<IGitRef> PriotizedBranches(IReadOnlyList<IGitRef> branches)
+        {
+            return OrderByPriority(branches, node => node.LocalName, AppSettings.PrioritizedBranchNames);
+        }
+
+        protected IEnumerable<RemoteRepoNode> PriotizedRemotes(IReadOnlyList<RemoteRepoNode> remotes)
+        {
+            return OrderByPriority(remotes.OrderBy(node => node.FullPath), node => node.FullPath, AppSettings.PrioritizedRemoteNames);
+        }
+
+        protected IEnumerable<Remote> PriotizedRemotes(IReadOnlyList<Remote> remotes)
+        {
+            return OrderByPriority(remotes.OrderBy(node => node.Name), node => node.Name, AppSettings.PrioritizedRemoteNames);
+        }
+
+        /// <summary>
+        /// Order the references by the prioties in the setting regex
+        /// </summary>
+        /// <typeparam name="T">The type to prioritize, e.g. IGitRef.</typeparam>
+        /// <param name="references">The branches or remotes to prioritize.</param>
+        /// <param name="keySelector">Function in T to get the sorter.</param>
+        /// <param name="setting">String with regexes with priorities separated by semicolon.</param>
+        /// <returns>The resorted references.</returns>
+        private IEnumerable<T> OrderByPriority<T>(IEnumerable<T> references, Func<T, string> keySelector, string setting)
+        {
+            // Sort prio branches first (if set) with the compile cache (no need to instanceiate)
+            var regexes = setting.Split(";", StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+            if (!regexes.Any())
+            {
+                return references;
+            }
+
+            int currPrio = 0;
+            Dictionary<string, int> prios = new();
+            foreach (string regex in regexes.Reverse())
+            {
+                currPrio++;
+                references.ForEach(node =>
```

Comment:
Suggest using `foreach` instead - it's faster as it has no additional layers of indirection.

Reply:
ok